c2rust-analyze/src/main.rs

```diff
@@ -222,6 +382,68 @@ fn describe_span(tcx: TyCtxt, span: Span) -> String {
     format!("{}: {}{}{}", line, src1, src2, src3)
 }
 
+/// Return all `body_owners`, ordered according to a postorder traversal of the graph of references
+/// between bodies.
+fn walk_callgraph(tcx: TyCtxt) -> Vec<LocalDefId> {
```

Comment:
Could be an iterator, external like `Iterator` or internal like `for_each_callee`.

Reply:
In the future (in code I've already written based on top of this branch), it seems like we'll need to iterate over the ordering multiple times, so I think leaving it as `Vec` is best.

Reply:
I think `walk` isn't the best name then.  How about `collect` or `walk_and_collect`?  `walk` sounds like an iterator (internal or external).

Reply:
OK, I've renamed it to `body_owners_postorder` (since it returns the same sequence as `tcx.body_owners()`, but sorted according to a postorder traversal)